### ISSUE TITLE
workflows in forks?

### ISSUE THREAD

**Issue by asmagill** [Hammerspoon/hammerspoon#2616]
@cmsj did you mean for the testing workflows to run in forks?

**Comment by asmagill** [comment]
Looks like something like they describe here might work: https://github.community/t/do-not-run-cron-workflows-in-forks/17636/2

**Comment by cmsj** [comment]
I kinda chose to not really take a view on it - it might be helpful for forks to just automatically inherit CI? I'm happy to be convinced either way since I have the luxury of mostly working on the org repos :)

**Comment by asmagill** [comment]
I guess I don’t mind the basic tests but I don’t know that the IRC channel needs to be updated unless its the main repo or pull request... and that step is failing on my fork anyways (haven’t looked into why but I presume a missing token or similar) so it would be better if that step failing didn’t trigger the “Tests Failed” email.

**Comment by cmsj** [comment]
That's a very good point actually, I'll check over the steps and see what makes sense to have be conditional.

(The IRC step is broken anyway, it needs a container to run, which doesn't work on macOS environments)

**Comment by cmsj** [comment]
In theory 180cc45d should both fix IRC notifications and limit them to the HS org's repo. Let's see :)

**Comment by asmagill** [comment]
@cmsj, It looks like the fix has at least prevented the errors from appearing in pushes to my repo's reports, but I am still consistently seeing a failure with HSscreen testSetMode:

~~~
Failing tests:
	Hammerspoon Tests:
		-[HSscreen testSetMode]
~~~

If this test shouldn't be being performed under CI control, and we do want to keep the tests running on forks, it should probably be suppressed like others that are known to have had issues with Travis before.

**Comment by cmsj** [comment]
d9a8d461 should disable that test in CI, thanks!

**Comment by asmagill** [comment]
That did work (after merging it with the latest master), and is why I merged #2555, but now I'm intermittently seeing:

~~~
Failing tests:
	Hammerspoon Tests:
		-[HSapplicationTests testInitWithPid]
~~~

which makes me ambivalent about performing the testing on forks, at least right now... on the one hand it's nice to know when things break, on the other hand if the tests themselves are brittle enough that we can't always trust them, its probably going to send the wrong message to people working on their own forks or preparing for submitting a future pull request...

Your thoughts?

**Comment by asmagill** [comment]
I guess if they are working on a true fork (and FWIW, I sort of expect one when we move to 2.0 and drop pre Big Sur support), then they can always remove the workflow, so it's more about people preparing a  branch for a future pull request.

**Comment by cmsj** [comment]
Yeah it's really tough. I'm kinda annoyed about the unreliable tests - I don't want to end up just skipping tons of stuff in CI, but  it's also super annoying, and I don't know why these things are unstable.
